### ISSUE TITLE
Unable to import manually created texture array

### ISSUE THREAD

**Issue by tinmanjuggernaut** [Zylann/godot_heightmap_plugin#273]
In the past, I've been able to import manually created texture arrays. Now I'm not. I just packed 2 4x3 arrays using the channel packer and set them up as arrays with mipmap/repeat and 4x3 for import.

When I click edit, change mode to textureArrays, and click Load Array, and select the file, it waits a second, then returns back to that dialog box. In the console I have this:

```
SCRIPT ERROR: _on_LoadTextureArrayDialog_file_selected: Invalid get index '0' (on base: 'PoolIntArray').
          At: res://addons/zylann.hterrain/tools/texture_editor/set_editor/texture_set_editor.gd:377
```

![image](https://user-images.githubusercontent.com/632766/126912669-c664081f-04e4-43c1-96ba-75ed3d378fb4.png)

Textures mode works fine and allows me to load 1-4 textures. If I change back to TextureArray, it asks if I want to unload all textures. Yes. I get the same thing.

Clicking clear gives me the same error on 382 and 390.

I had the system create packed textures successfully, with nice blending using multi16, but I don't want to include cached tex imports in my git repo.

**Environment**
win 10/64, gtx1060

**Comment by Calinou** [comment]
Related to https://github.com/godotengine/godot/issues/34312?

**Comment by tinmanjuggernaut** [comment]
I looked at that ticket, but texturearrays loaded fine before. I think this is an hterrain bug, as it's erroring out due to an empty array in the UI. The texture array loads, according to the assert. It fails on the next line when its querying the UI array.

Here it's loading the file. 382 and 390 are clearing the UI, but have a similar line.

```
func _on_LoadTextureArrayDialog_file_selected(fpath: String):
	assert(_texture_set.get_mode() == HTerrainTextureSet.MODE_TEXTURE_ARRAYS)
	var texture_array = load(fpath)
	assert(texture_array != null)
	var slot_index : int = _slots_list.get_selected_items()[0]
	_set_texture_array_action(slot_index, texture_array, _load_texture_type)
```

Those other tools create a wrapper so you don't have to manually create a texture array. But I've already made it, I just need to load it.

**Comment by Zylann** [comment]
Since when this started being an issue? The plugin hasn't received significant updates since january.

**Comment by tinmanjuggernaut** [comment]
I haven't tried using TextureArrays in months. I've been using my version of classic4. However I've been replacing all of my assets and textures and now I'm testing TAs again.

I was able to work around the UI bug by having the importer creating a throwaway texture array, saving the texture set resource as a .tres, and editing the file to point to my PNG texture arrays. Reloading the scene I now have all 12 channels.

Looking at the edit textures window, I now see the individual layers. I think the problem is that if you switch to TextureArrays,  you no longer have the +/- buttons to add a slot, and one isn't added by default, so `_slots_list` is always empty.

Also @Zylann the blending is really good on Multisplat16. Good job!

**Comment by Zylann** [comment]
Looking at the code, it might be a mistake because `slot_index` would then map to the `ground_texture_type` argument in `HTerrainTextureSet.set_texture_array`. It makes sense for the "textures" mode but not for "texture array".

**Comment by Zylann** [comment]
I posted a fix in 261a37659bd2b49d38462e434b1a9a61164aaf4c, let me know if it works for you

**Comment by tinmanjuggernaut** [comment]
Yep, works well. Thanks for the help.
